backend/settings.go

```diff
@@ -58,7 +60,10 @@ func (m *Backend) GetPool() int {
 		if m.GetTestnet() {
 			return 2 // Default P2Pool on testnet
 		}
-		return 3 // Default Hashalot on mainnet (for now...)
+		// Default to a random pool
+		rand.Seed(time.Now().UnixNano())
+		pools := pools.GetPools(m.Address(), m.GetTestnet())
+		return pools[rand.Intn(len(pools))].GetID()
```

Comment:
If you randomize it here, you will have to save it too before returning it - ensuring that the user will stay on that pool. 

This code ensures that you return a default if none is set, and we can change that default in version updates. But the setting will remain unset so it'll return whatever's the default in a next release. 

This means that you'll switch pools between restarts if you don't save it after determining the (random) default, potentially leaving behind partial reward payouts. I think if we randomize it we can just fix new users to whatever's chosen as the random default.

Reply:
I agree, it's safest to save the "pool" setting as soon as the random pool is selected. Just didn't know if that was a kosher thing to be doing within a getter method, so we had put the save in the `backend.ResetPool()` function instead (which seemed to work -- the randomly-selected pool setting did stick). But if you think it's right to do the saving immediately after the random pool is selected, let's do that!

Change is here: https://github.com/ehusby/one-click-miner-vnext/commit/2e6b668a0df272f12858f7d54b7131d86e5ba0e8
I'll poke @vertiond to update this PR with this change in the morning.